starlette/_compat.py

```diff
@@ -27,3 +32,9 @@ def md5_hexdigest(
 
     def md5_hexdigest(data: bytes, *, usedforsecurity: bool = True) -> str:
         return hashlib.md5(data).hexdigest()
+
+
+try:
+    from contextlib import aclosing  # type: ignore[attr-defined]
+except ImportError:  # Python < 3.10
+    from async_generator import aclosing  # type: ignore
```

Comment:
```suggestion
# see contextlib.aclosing
class aclosing:
    def __init__(self, thing: AbstractAsyncContextManager):
        self.thing = thing
 
    async def __aenter__(self):
        return self.thing
   
    async def __aexit__(self, *exc_info):
        await self.thing.aclose()
```

Reply:
(need the import for `AbstractAsyncContextManager`, or it could just be `Any` honestly)